### ISSUE TITLE
IDPEP and FDR crashes after ConsensusID

### ISSUE THREAD

**Issue by lars20070** [OpenMS/OpenMS#1428]
IDPEP crashes on the following idXML [1] with

"No data collected. Check whether search engine is supported."

The IDs come from several fixed MSGF+ searches combined using ConsensusID. ConsensusID followed by IDPEP seems to be a problem.

[1] https://www.dropbox.com/sh/7xo5k0pfscz5i3r/AAADnETdtfJfeviG-guwrTrRa

**Comment by lars20070** [comment]
@jpfeuffer @hendrik "No data collected" appears when all scores are empty. Why are there no scores after combining MSGF+ search results with ConsensusID?

**Comment by jpfeuffer** [comment]
The initial issue is easily explained. IDPEP currently does not support ConsensusID as a "search engine".
We would need to decide on which score in the PeptideHit to use, when the search engine in the IdentificationRun entry is "OpenMS/ConsensusID_...", then it should be a quick fix.
Your file e.g. still carries FDRs as scores.
I need to look into ConsensusID a bit. I don't know how it uses the old PeptideHit scores and how/if it updates them. If its output is only the additional consensus_id_support metavalue, then something else went wrong in your example as all are "nan" or "0".
Maybe that is what you mentioned in your comment, however I didn't get what happened before this comment. Why did the warning change? What did you do to get it to work this far?

**Comment by lars20070** [comment]
@jpfeuffer Just to clarify the workflow:
(1) Three separate MSGF+ fixed-mod searches
(2) Three separate FDR calculation
(3) IDMerger
(4) ConsensusID best
(5) PeptideIndexer
(6) IDPEP -> crash
Is it not allowed to use q-values (from FDR) instead of the original MSGF+ scores in ConsensusID best?

**Comment by timosachsenberg** [comment]
why should one do the FDR calculation separatly?

**Comment by hendrikweisser** [comment]
@jpfeuffer:
> I need to look into ConsensusID a bit. I don't know how it uses the old PeptideHit scores and how/if it updates them.

That depends on the algorithm used, but should be documented reasonably well. In all algorithms except "ranks", the output score type is the same as the input score type.

> If its output is only the additional consensus_id_support metavalue, then something else went wrong in your example as all are "nan" or "0".

There is more going on. "consensus_id_support" is just an informative value.

@lars20070:
> (1) Three separate MSGF+ fixed-mod searches

Were these done with the adapter or on the command line? ConsensusID needs RT information to work.

> Is it not allowed to use q-values (from FDR) instead of the original MSGF+ scores in ConsensusID best?

It is allowed, but I don't think it makes as much sense as using the MS-GF+ scores.

@timosachsenberg:
> why should one do the FDR calculation separatly?

I agree, one probably shouldn't.

**Comment by hendrikweisser** [comment]
I've now had a look at the ConsensusID output file that Lars linked above. Seems good to me. RT information was present and ConsensusID grouped the IDs from the three searches successfully (by spectrum). The "consensus_id_support" value is zero because the searches were done with different modifications, so there aren't any common hits between them. The exception are the "nan" cases, where there was only one search hit for a spectrum, so no comparison was possible.

**Comment by jpfeuffer** [comment]
Okay. Then it is the mentioned IDPEP problem, that has no if-case for ConsensusID "search engines".
The problem is, that the score type can be basically everything right?
E.g. e-values have to be log transformed, other don't.

**Comment by hendrikweisser** [comment]
> The problem is, that the score type can be basically everything right?

In the case of the "best" and "average" algorithms, yes.

I'm not sure if it even makes sense to support ConsensusID results in IDPEP. In my opinion IDPEP should be run before ConsensusID.

**Comment by lars20070** [comment]
> On 26 May 2015, at 11:37, Timo Sachsenberg <notifications@github.com> wrote:
> 
> why should one do the FDR calculation separatly?
> 

This was a test. We wanted to use search-engine-independent scores (q-values) to resolve the fixed-mod search results.

**Comment by lars20070** [comment]
> @lars20070:
> 
> (1) Three separate MSGF+ fixed-mod searches
> 
> Were these done with the adapter or on the command line? ConsensusID needs RT information to work.
> 

We used the MSGFPlusAdapter.

**Comment by lars20070** [comment]
This screenshot demonstrates what we are currently struggling with. MSGF+ peptide sequences differ from light, medium and heavy search. And the mods clash with the (rather clear) quants.

Since we did not get good results ConsensusID-combining MSGF+ searches based on the MSGF+ score, we tried it with the q-values.

![stillconflicts](https://cloud.githubusercontent.com/assets/6149805/7810933/9c169dec-03a6-11e5-9647-8bb1dcc5359c.png)

**Comment by hendrikweisser** [comment]
> The screenshot below demonstrates what we are currently struggling with.

I don't get what your picture is supposed to show. You have three features with three different IDs (light/medium/heavy). Where is the problem?

**Comment by jpfeuffer** [comment]
I think you have to somehow do ConsensusID on the unmodified peptide sequences.
Otherwise you won't get a consensus on the possible candidates of the consensus feature. They will always differ by their labelling.
If you are doing that already, have you checked if there are consensus candidates at all (i.e. PSMs with the same unmodified sequence across all labellings) that one could find?

**Comment by lars20070** [comment]
Here an example without conflicts. (Dimethyl) on the light quant, (Dimethyl:2H(4)) on the medium quant, (Dimethyl:2H(6)13C(2)) on the heavy quant. The same sequence MLRP... on each of the three quants.

Both screenshots come from the same dataset and analysis. We would like to have more IDs as seen in the second screenshot.

![noconflict](https://cloud.githubusercontent.com/assets/6149805/7812020/049a0e00-03af-11e5-9668-e11719cbbd36.png)

**Comment by lars20070** [comment]
@jpfeuffer The MultiplexResolver #1441 will take care of any ID / quant conflicts. But first we need to get the best possible IDs (as seen in the second screenshot).

**Comment by hendrikweisser** [comment]
> But first we need to get the best possible IDs (as seen in the second screenshot).

In the first screenshot, do I understand correctly that you would consider the middle ID as "correct", since that has the appropriate (medium) label? I don't think ConsensusID will help you in that case, since the IDs don't come from the same spectrum.

**Comment by lars20070** [comment]
@hendrikweisser That's exactly right. Only the medium ID is correct.

All IDs in the screenshots are already resolved (i.e. the ConsensusID output). Straight out of MSGF+ or after 5% FDR we still have multiple peptide hits for the same MS2 spectrum.

**Comment by timosachsenberg** [comment]
ok so this will be handled in MultiPlexResolver am I right?

**Comment by lars20070** [comment]
Think we got side-tracked. Back to the issue.

Either allow IdentificationRun is "OpenMS/ConsensusID_..." or modify the error message. See @jpfeuffer's first comment.

**Comment by lars20070** [comment]
Any updates @jpfeuffer? - Running IDPEP after ConsensusID would simplify many of our workflows.

**Comment by jpfeuffer** [comment]
I agree with @hendrikweisser's former comment, that it makes more sense to calculate FDR or PEPs before using ConsensusID. I actually looked at the distributions in the example file from you and they were useless for fitting. "Consensus" FDRs just do not fulfill the required assumptions.
There are even special methods for combining PEPs in ConsensusID. The scoretype PEP should not get lost during ConsensusID. You can still use Fido after that if that is your concern.

How should a better error message look like?

I will send you an email about how I would structure the above mentioned problematic workflow.

**Comment by lars20070** [comment]
Here a similar crash. `FalseDiscoveryRate` crashes with `No data collected. Check whether search engine is supported.` [Peptide identification workflow.](https://github.com/OpenMS/OpenMS/files/49961/crash.pdf) IMO, such a workflow should run through.

**Comment by jpfeuffer** [comment]
Ok. So both FDR and IDPEP should support ConsensusID_* as "search engine"?

**Comment by lars20070** [comment]
And `ConsensusID` needs to use the "right" score. https://github.com/OpenMS/OpenMS/issues/1691#issuecomment-155796462 See related discussion.

**Comment by hendrikweisser** [comment]
I get why IDPEP fails, but not FDR (tool). Shouldn't the score orientation ("higher_score_better") together with target/decoy information be enough to calculate the FDR?

**Comment by timosachsenberg** [comment]
Might be resolved. Please open if still relevant
